app/routes.py

```diff
@@ -220,44 +220,33 @@ def tower_settings(tower_id):
         print("form half_muffled is", form.half_muffled.data)
         tower.half_muffled = form.half_muffled.data
 
+        def emit_socket(name, data):
+            socketio.emit(name, data, broadcast=True, room=tower.tower_id)
+
         # ===== DEAL WITH WHEATLEY POTENTIALLY BEING ENABLED OR DISABLED =====
         wheatley_enabled = form.wheatley_enabled.data
         # Add this setting to the database's representation of the tower
         tower_db.wheatley_enabled = wheatley_enabled
         # If the enabledness of Wheatley has changed either way, broadcast that to the users of
         # that tower
         if tower.wheatley.enabled != wheatley_enabled:
-            socketio.emit('s_set_wheatley_enabledness', {'enabled': wheatley_enabled},
-                          broadcast=True, room=tower.tower_id)
+            emit_socket('s_set_wheatley_enabledness', {'enabled': wheatley_enabled})
             tower.wheatley.set_enabledness(wheatley_enabled)
             if wheatley_enabled:
                 # If Wheatley is being *enabled*, then update the setting in the tower and tell the
                 # clients that Wheatley has arrived
-                socketio.emit(
+                emit_socket(
                     's_user_entered',
                     {
                         'user_name': WHEATLEY_USER_NAME,
                         'user_id': WHEATLEY_USER_ID
                     },
-                    broadcast=True,
-                    include_self=True,
-                    room=tower.tower_id
                 )
-                socketio.emit('s_wheatley_setting', tower.wheatley.settings)
-                socketio.emit('s_wheatley_row_gen', tower.wheatley.row_gen)
+                emit_socket('s_wheatley_setting', tower.wheatley.settings)
+                emit_socket('s_wheatley_row_gen', tower.wheatley.row_gen)
```

Comment:
Is the main change that these previously were being broadcast to everyone on the server instead of a specific tower id?

Reply:
Yep sorry I should have explained but I did this on stream so everyone watching had the context for what was happening.  It was a pretty silly mistake but you can see why I didn't pick it up; I only created one room at a time when testing.